skimage/draw/draw.py

```diff
@@ -477,8 +485,20 @@ def polygon(r, c, shape=None):
            [0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
 
+    >>> from skimage.draw import polygon
+    >>> shape = (4, 4)
+    >>> img = np.zeros(shape, dtype=np.uint8)
+    >>> r = np.array([1, 2, 8])
+    >>> c = np.array([1, 7, 4])
+    >>> rr, cc = polygon(r, c, shape=shape, offset=(2, 2))
+    >>> img[rr, cc] = 1
+    >>> img
+    array([[1, 1, 1, 1],
+           [1, 1, 1, 1],
+           [0, 1, 1, 1],
+           [0, 1, 1, 1]], dtype=uint8)
     """
-    return _polygon(r, c, shape)
+    return _polygon(r, c, shape, offset)
```

Comment:
Thanks @henrypinkard for the contribution!

I'm not sure if I'm missing something but isn't there a way simpler solution to achieve the behavior you describe?
```suggestion
    r = np.atleast_1d(r)
    c = np.atleast_1d(c)
    if offset is not None:
        r -= offset[0]
        c -= offset[1]
    return _polygon(r, c, shape)
```
I haven't tested it thoroughly yet but it seems to do the trick, passes your tests and even supports negative offsets. It would be more straightforward to understand and maintain as well. I prefer to keep Cython code as simple and minimal as possible, because it's way harder to debug sometimes.

We may even want to refrain form adding a new parameter to `polygon` at all (but keep it in `polygon2mask`)  and document how to do this with a good docstring/example...